library/ssl_tls.c

```diff
@@ -2872,6 +2866,8 @@ int ssl_parse_certificate( ssl_context *ssl )
             if( pk_can_do( pk, POLARSSL_PK_ECKEY ) &&
                 ! ssl_curve_is_acceptable( ssl, pk_ec( *pk )->grp.id ) )
             {
+                ssl->session_negotiate->verify_result |= BADCERT_BAD_EC;
```

Comment:
@andresag01 There's no preexisting equivalent to `MBEDTLS_X509_BADCERT_BAD_KEY` in mbed TLS 1.3, hence the addition of a new error code.